OpenOversight/tests/test_commands.py

```diff
@@ -79,66 +106,639 @@ def test_add_job_title__success(session):
+
+    n_created, n_updated = bulk_add_officers([csvfile], standalone_mode=False)
+    assert n_created == 0
+    assert n_updated == 1
+
+    officer = Officer.query.filter_by(unique_internal_identifier=officer_uid).one()
+
+    assert officer.first_name == "FOO"
+
+
+def test_csv_new_officer(csvfile):
+    df = pd.read_csv(csvfile)
+
+    n_rows = len(df.index)
+    assert n_rows > 0
+
+    n_officers = Officer.query.count()
+    assert n_officers > 0
+
+    new_uid = str(uuid.uuid4())
+    new_officer = {  # Must match fields in csvfile
+        "department_id": 1,
+        "unique_internal_identifier": new_uid,
+        "first_name": "FOO",
+        "last_name": "BAR",
+        "middle_initial": None,
+        "suffix": None,
+        "gender": "F",
+        "race": "BLACK",
+        "employment_date": None,
+        "birth_year": None,
+        "star_no": 666,
+        "job_title": "CAPTAIN",
+        "unit": None,
+        "star_date": None,
+        "resign_date": None,
+        "salary": 1.23,
+        "salary_year": 2019,
+        "salary_is_fiscal_year": True,
+        "overtime_pay": 4.56,
+    }
+    df = df.append([new_officer])
+    df.to_csv(csvfile)
+
+    n_created, n_updated = bulk_add_officers([csvfile], standalone_mode=False)
+    assert n_created == 1
+    assert n_updated == 0
+
+    officer = Officer.query.filter_by(unique_internal_identifier=new_uid).one()
+
+    assert officer.first_name == "FOO"
+    assert Officer.query.count() == n_officers + 1
+
+
+def test_csv_new_salary(csvfile):
+    # Delete all current officers and salaries
+    Salary.query.delete()
+    Officer.query.delete()
+
+    assert Officer.query.count() == 0
+
+    df = pd.read_csv(csvfile)
+    df.loc[0, "salary"] = "123456.78"
+    df.to_csv(csvfile)
+
+    n_created, n_updated = bulk_add_officers([csvfile], standalone_mode=False)
+    assert n_created > 0
+    assert n_updated == 0
+    officer_count = Officer.query.count()
+    assert officer_count == n_created
+
+    officer = get_officer(
+        1, df.loc[0, "star_no"], df.loc[0, "first_name"], df.loc[0, "last_name"]
+    )
+    assert officer
+    officer_id = officer.id
+    assert len(list(officer.salaries)) == 1
+
+    # Update salary
+    df.loc[0, "salary"] = "150000"
+    df.to_csv(csvfile)
+
+    assert Officer.query.count() > 0
+    n_created, n_updated = bulk_add_officers([csvfile], standalone_mode=False)
+    assert n_created == 0
+    assert n_updated == 1
+    assert Officer.query.count() == officer_count
+
+    officer = Officer.query.filter_by(id=officer_id).one()
+    assert len(list(officer.salaries)) == 2
+    for salary in officer.salaries:
+        assert float(salary.salary) == 123456.78 or float(salary.salary) == 150000.00
+
+
+# NEW TESTS
+# --update_by_name && two with same name
+# --update_by_name && success
+# update null static field with and without --update-static-fields
+
+
+def test_bulk_add_officers__success(session, department_with_ranks, csv_path):
```

Comment:
This test looks like it is for the base case, and seems redundant with the `test_csv_import` tests, but not necessarily problematic. @abandoned-prototype is there something this test does that the older ones don't?

Reply:
This test is for the base-case aka "happy path" of using the flag `--update-by-name`. This would fail if the flag wasn't set, so I don't think there is any other test that tests this scenario